### PR TITLE
[TrilinosApplication] Minor changes from #10734

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Minor changes from #10734 in order to reduce size of #10734

**🆕 Changelog**

- [Replace typedef with using](https://github.com/KratosMultiphysics/Kratos/commit/0c067d63a435a35885f69906f01f61a7ae307521)
- [Clean up the code a bit](https://github.com/KratosMultiphysics/Kratos/commit/73a17478c5092b425d7d9c700a2972e17e8bfc2c)
- [Improve TrilinosCPPUtilities](https://github.com/KratosMultiphysics/Kratos/commit/504a8e753a3e8aa2c34393bf0083c0bd7294f7c8)
- [Move TrilinosCPPUtilities to cpp](https://github.com/KratosMultiphysics/Kratos/commit/5bb7c2d26f2800abca992c63450fed5ffd5ce550)

### CODE REVIEW

File: applications/TrilinosApplication/custom_utilities/trilinos_cpp_test_utilities.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                num_non_diagonal_entries = 1;
+            } else if (MyGlobalElements[i] == NumGlobalElements-1) {
+                non_diagonal_indices[0] = NumGlobalElements-2;
+                num_non_diagonal_entries = 1;
+            } else {
+                non_diagonal_indices[0] = MyGlobalElements[i]-1;
+                non_diagonal_indices[1] = MyGlobalElements[i]+1;
+                num_non_diagonal_entries = 2;
+            }
+            A.InsertGlobalValues(MyGlobalElements[i], num_non_diagonal_entries, non_diagonal_values, non_diagonal_indices);
```
Comment (philbucher) [inline applications/TrilinosApplication/custom_utilities/trilinos_cpp_test_utilities.cpp]:
```suggestion
    std::vector<double> non_diagonal_values(2);
    non_diagonal_values[0] = -1.0; non_diagonal_values[1] = -1.0;
    std::vector<int> non_diagonal_indices(2);
    int num_non_diagonal_entries;
    double value;
    for( int i=0 ; i<NumMyElements; ++i ) {
        // Non diagonal values
        if (AddNoDiagonalValues) {
            if (MyGlobalElements[i]==0) {
                non_diagonal_indices[0] = 1;
                num_non_diagonal_entries = 1;
            } else if (MyGlobalElements[i] == NumGlobalElements-1) {
                non_diagonal_indices[0] = NumGlobalElements-2;
                num_non_diagonal_entries = 1;
            } else {
                non_diagonal_indices[0] = MyGlobalElements[i]-1;
                non_diagonal_indices[1] = MyGlobalElements[i]+1;
                num_non_diagonal_entries = 2;
            }
            A.InsertGlobalValues(MyGlobalElements[i], num_non_diagonal_entries, non_diagonal_values.data(), non_diagonal_indices.data());
```

File: applications/TrilinosApplication/trilinos_space.h [hunk 2]
```diff
@@ -307,12 +323,15 @@ class TrilinosSpace
...
+        const MatrixType& rA,
+        const VectorType& rX,
+        VectorType& rY
+        )
     {
-        //y = A*x
-        bool transpose_flag = true;
+        constexpr bool transpose_flag = true;
         rA.Multiply(transpose_flag, rX, rY);
-    } // rY = rAT * rX
```
Comment (philbucher) [inline applications/TrilinosApplication/trilinos_space.h:315]: I find this one very useful, can you add it to the fct documentation please?

**Review by philbucher — COMMENTED** [review]
minor, otherwise good

File: applications/TrilinosApplication/trilinos_space.h (hunk 2 above)
Comment (loumalouomega) [reply applications/TrilinosApplication/trilinos_space.h:315]: Okay

Comment (loumalouomega) [reply applications/TrilinosApplication/trilinos_space.h:315]: It is already in the function documentation!

**Comment by loumalouomega** [comment]
Ready @philbucher

**Review by philbucher — DISMISSED** [review]

**Comment by loumalouomega** [comment]
It is me or the pipelines are stuck? @roigcarlo

**Comment by loumalouomega** [comment]
> It is me or the pipelines are stuck? @roigcarlo

@philbucher I needed to trigger the CI

**Comment by roigcarlo** [comment]
Seems that lately there are problems with the triggers, I had to make a couple empty commits last week for the same reason. :S

**Comment by loumalouomega** [comment]
> Seems that lately there are problems with the triggers, I had to make a couple empty commits last week for the same reason. :S

Looks like

**Review by philbucher — APPROVED** [review]

**Pull request merged**